src/wrapper.rs

```diff
@@ -162,22 +161,22 @@ impl<'a> DxcIncludeHandlerWrapper<'a> {
 
         let filename = crate::utils::from_wide(filename as *mut _);
 
-        let source = unsafe { &(*me).handler.load_source(filename) };
+        let source = unsafe { (*me).handler.load_source(filename) };
 
         if let Some(source) = source {
-            let pinned_source = Rc::new(source.clone());
+            let pinned_source = Rc::new(source);
```

Comment:
Isn't this where we should use a `Pin<Rc<String>>`?

Reply:
Probably yeah, I'm not sure 😊

Reply:
Yeah this makes me curious how it works. The documentation really only specifies self-referential structs and moving of that object, but that is of no relevance here: even if the `String` moves, the underlying slice on the heap is unlikely to move.

However, `Pin` works on _pointer types_. `String` is a "pointer" of type `&str` to some owned data, and this is clear when interacting with a `Pin<String>`: it will only ever give the pointer type `&str` back meaning we cannot do any disastrous modifications like `.reserve` (which usually reallocates) that are only available on the `String` type (whenever we have mutable access to it). In other words, it appears `Pin` takes over the ownership for us and only ever allows us to retrieve the value of the underlying pointer.

Note that I should have been more rigorous in cleaning the `Rc` up too: this value is not ever borrowed more than once and hence has no effect (except that the value is technically borrowed a second time inside `IDxcBlobEncoding`, but Rust does not know that and cannot model it as the object exposing that interface is owned by DXC).